src/main/java/org/myrobotlab/service/Python.java

```diff
@@ -561,19 +600,25 @@ public void openScriptFromFile(String filename) throws IOException {
     openScript(filename, data);
   }
 
-  public void onRegistered(Registration r) {
-    ServiceInterface s = r.service;
+  public void onStarted(String serviceName) {

```

Comment:
this feels really odd to me.. this stuff should be done in startService ...  not in the constructor. and not in a call back like  onStarted... no?